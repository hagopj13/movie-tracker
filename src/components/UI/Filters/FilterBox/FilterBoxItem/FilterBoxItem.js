// @flow
import React from 'react';
import type { Node } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

type Props = {
  children: Node,
  title?: string,
};

const useStyles = makeStyles((theme) => ({
  root: {
    padding: theme.spacing(2),
  },
  title: {
    marginBottom: theme.spacing(1),
  },
}));

const FilterBox = (props: Props) => {
  const { children, title } = props;

  const classes = useStyles();

  return (
    <div className={classes.root}>
      {title && <Typography className={classes.title}>{title}</Typography>}
      {children}
    </div>
  );
};

FilterBox.defaultProps = {
  title: null,
};

export default FilterBox;
