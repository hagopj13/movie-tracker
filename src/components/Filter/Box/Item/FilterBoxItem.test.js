import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Typography from '@material-ui/core/Typography';

import FilterBoxItem from './FilterBoxItem';

describe('FilterBoxItem component', () => {
  let shallow;
  let wrapper;

  const ChildComponent = () => <div>Child component</div>;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    wrapper = shallow(
      <FilterBoxItem>
        <ChildComponent />
      </FilterBoxItem>,
    );
  });

  it('should render the FilterBoxItem component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the child component', () => {
    expect(wrapper.exists(ChildComponent)).toBe(true);
  });

  it('should render the FilterBoxItem component correctly if title prop is specified', () => {
    const title = 'Some title';
    const mockProps = { title };
    const newWrapper = shallow(
      <FilterBoxItem {...mockProps}>
        <ChildComponent />
      </FilterBoxItem>,
    );
    expect(newWrapper).toMatchSnapshot();
  });

  it('should render the title if title prop is specified', () => {
    const title = 'Some title';
    const mockProps = { title };
    const newWrapper = shallow(
      <FilterBoxItem {...mockProps}>
        <ChildComponent />
      </FilterBoxItem>,
    );
    expect(newWrapper.find(Typography).text()).toBe(title);
  });
});
