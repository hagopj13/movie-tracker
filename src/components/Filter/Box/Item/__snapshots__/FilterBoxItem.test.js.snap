// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBoxItem component should render the FilterBoxItem component correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <ChildComponent />
</div>
`;

exports[`FilterBoxItem component should render the FilterBoxItem component correctly if title prop is specified 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-title-2"
  >
    Some title
  </WithStyles(ForwardRef(Typography))>
  <ChildComponent />
</div>
`;
